src/ext/java/org/opentripplanner/ext/flex/FlexPathDurations.java

```diff
@@ -0,0 +1,80 @@
+package org.opentripplanner.ext.flex;
+
+import org.opentripplanner.framework.time.DurationUtils;
+
+/**
+ * This value-object contains the durations for a Flex access or egress path. The path may also
+ * contain access and egress "parts" - a walk part in the beginning and/or at the end. It has
+ * method for mapping between the flex-trip-times and the router(OTP) time.
+ * <p>
+ * If the {@code access/egress} is empty the duration is zero. The flex-trip duration must be
+ * at least 1 second.
```

Comment:
This requirement is not the same as MIN_FLEX_TRIP_DURATION_SECONDS but I'm not sure if they are supposed to be.

Reply:
The precondition in the constructor are there to prevent illegal objects from being created - it protect from programming errors and side-effects witch might happen at a later stage. It is not enforcing business rules. The `FlexPathDuration` state its contract - the flex-trip-duration must be at least 1s; Hence all other code using it can trust that and do not have to do any sanity checks. 

The business rule that a Flex trip need to be at least 10 seconds is implemented in on place, in other places you should assume that a flex trip can be 1s - this make the code more robust and we can change the business rule later.

We could put the business rule (>=10s) into the `FlexPathDurations`, but it would be a bit hidden away and it would need to be static method, since we do the check before we create the `FlexPathDurations`.

This is also known as "syntax vs semantic validation".